docs (Microsoft Auth): Add docs regarding whitelist system (#300)
dscalzi/HeliosLauncher: docs/MicrosoftAuth.md

    - Select **Mobile and desktop applications**.
    - Choose `https://login.microsoftonline.com/common/oauth2/nativeclient` as the **Redirect URI**.
    - Select **Configure** to finish adding the platform.
8. Go to **Credentials & secrets**.
    - Select **Client secrets**.
    - Click **New client secret**.
    - Set a description.
    - Click **Add**.

>     - Don't copy the client secret, adding one is just a requirement from Microsoft, for some reason.

Reviewer: ```suggestion
    - Don't copy the client secret, adding one is just a requirement from Microsoft.
```

We shouldnt sound annoyed in the docs

8. Navigate back to **Overview**.
9. Copy **Application (client) ID**.

## Adding the Azure Client ID to Helios Launcher.

In `app/assets/js/ipcconstants.js` you'll find **`AZURE_CLIENT_ID`**. Set it to your application's id.

Note: Azure Client ID is NOT a secret value and **can** be stored in git. Reference: https://stackoverflow.com/questions/57306964/are-azure-active-directorys-tenantid-and-clientid-considered-secrets

> Then relaunch your app, and login. You'll be greeted with an error message, because the app isn't whitelisted yet. And for some reason, Microsoft needs some activity on the app before whitelisting it. __Trying to log in before requesting whitelist is mandatory.__

Reviewer: ```suggestion
Then relaunch your app, and login. You'll be greeted with an error message, because the app isn't whitelisted yet. Microsoft needs some activity on the app before whitelisting it. __Trying to log in before requesting whitelist is mandatory.__
```

## Requesting whitelisting from Microsoft

1. Ensure you have completed every step of this doc page.

> 2. Fill [this form](https://aka.ms/mce-reviewappid) with required information. Remember this is a new appID for approval. You can find both the Client ID and the Tenant ID on the overview page in the Azure Portal.

Reviewer: sorry I edited my correction but you were too fast lol

```suggestion
2. Fill [this form](https://aka.ms/mce-reviewappid) with the required information. Remember this is a new appID for approval. You can find both the Client ID and the Tenant ID on the overview page in the Azure Portal.
```

> 2. Fill [this form](https://aka.ms/mce-reviewappid) with required informations. Remember this is a new appID for approval. You can find both the Client ID and the Tenant ID on the overview page in the Azure Portal.

Reviewer: ```suggestion
2. Fill [this form](https://aka.ms/mce-reviewappid) with required information. Remember this is a new appID for approval. You can find both the Client ID and the Tenant ID on the overview page in the Azure Portal.
```

3. Give Microsoft some time to review your app.

> 4. Once receiving Microsoft's approval, wait up to 24 hours to allow changes to apply.

Reviewer: ```suggestion
4. Once you have received Microsoft's approval, allow up to 24 hours for the changes to apply.
```
